[cxx-interop] Temporarily exclude CxxSetToCollection benchmark from the build

egorzhdan opened the pull request:
This is needed to unblock nightly toolchains, which are currently failing to build with an assertion:
```
swift-frontend: /home/build-user/build/buildbot_linux/llvm-linux-x86_64/tools/clang/include/clang/AST/TypeNodes.inc:79: TypeInfo clang::ASTContext::getTypeInfoImpl(const Type *) const: Assertion `!T->isDependentType() && "should not see dependent types here"' failed.
```

rdar://150067288

<!--
If this pull request is targeting a release branch, please fill out the
following form:
https://github.com/swiftlang/.github/blob/main/PULL_REQUEST_TEMPLATE/release.md?plain=1

Otherwise, replace this comment with a description of your changes and
rationale. Provide links to external references/discussions if appropriate.
If this pull request resolves any GitHub issues, link them like so:

  Resolves <link to issue>, resolves <link to another issue>.

For more information about linking a pull request to an issue, see:
https://docs.github.com/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

j-hui reviewed (commented):
There were a couple others failing too, including `VectorSum` and `StringConversion`. Those should probably be commented out too.

egorzhdan committed:
[cxx-interop] Temporarily exclude CxxSetToCollection benchmark from the build

This is needed to unblock nightly toolchains, which are currently failing to build with an assertion:
```
swift-frontend: /home/build-user/build/buildbot_linux/llvm-linux-x86_64/tools/clang/include/clang/AST/TypeNodes.inc:79: TypeInfo clang::ASTContext::getTypeInfoImpl(const Type *) const: Assertion `!T->isDependentType() && "should not see dependent types here"' failed.
```

rdar://150067288

egorzhdan commented:
@swift-ci please benchmark

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@j-hui good point! I disabled the other two, let's see if anything else is failing.

j-hui reviewed (approved):
This looks good to me, but I'm wondering why you're commenting out the contents of the benchmark rather than commenting out the cmake entry (as you were doing originally)?

j-hui commented:
@swift-ci please smoke test macos platform

j-hui commented:
@swift-ci please smoke test macos platform

MaxDesiatov commented:
@swift-ci smoke test macos

MaxDesiatov commented:
@swift-ci smoke test macos

egorzhdan commented:
> This looks good to me, but I'm wondering why you're commenting out the contents of the benchmark rather than commenting out the cmake entry (as you were doing originally)?

@j-hui the benchmarks can be built as a Swift Package, in which case CMake flags won't be applied. This way we can be certain that the problematic code won't be built regardless of the build system used.

MaxDesiatov commented:
@egorzhdan does this need a cherry-pick for 6.2?
